### PR TITLE
feat: Meta-data. Fixes #161 

### LINKED ISSUE #161: Message meta-data
Currently messages are just byte-arrays. This the lowest common-denominator, but means that dataflow can't infer useful information about messages. 

Use case for meta-data include:

* Tracing messages through the system to diagnose problem
* Message de-duplication
* Grouping by time
* Verifying messages

What sources already provide:

* HTTP:
  * Etag
  * Content-length
* Kafka:
  * Timestamp
  * Topic
  * Partition
  * Offset
* S3:
  * Etag 
  * Last-modified time
  * Size
  * Owner
* STAN:
  * Sequence
  * Timestamp
  * Redelivery
  * Subject

Cloud-events provides a formal specification:

* ID (UUID)
* Source (URI)
* Type
* Data-content-type
* Dataschema
* Subject
* Time

```json
{
    "specversion" : "1.0",
    "type" : "com.github.pull_request.opened",
    "source" : "https://github.com/cloudevents/spec/pull",
    "subject" : "123",
    "id" : "A234-1234-1234",
    "time" : "2018-04-05T17:31:00Z",
    "comexampleextension1" : "value",
    "comexampleothervalue" : 5,
    "datacontenttype" : "text/xml",
    "data" : "<much wow=\"xml\"/>"
}
```

### PR DESCRIPTION
**Pull request by alexec**
Fixes #161

### CODE REVIEW

**Comment by alexec** [comment]
@whynowy @vigith as per the [Ship/Show/Ask](https://martinfowler.com/articles/ship-show-ask.html) convention this is "Ask".

This PR adds meta-data, specifically `source` and `id` that allows us to uniquely identify a message *within a step* and potentially allow us to send that `id` downstream to other steps.

Let's chat next week.

File: runner/sidecar/out.go [hunk 1]
```diff
@@ -37,7 +39,10 @@ func connectOutHTTP(sink func(context.Context, []byte) error) {
 			_, _ = w.Write([]byte(err.Error()))
 			return
 		}
-		if err := sink(r.Context(), data); err != nil {
+		if err := sink(
+			dfv1.ContextWithMeta(r.Context(), fmt.Sprintf("%s.pod.%s.%s", pod, namespace, cluster), uuid.New().String()),
```
Comment (whynowy) [inline runner/sidecar/out.go:43]: The `source` and `ID` are changed when they are sent to downstream?

**Review by whynowy — COMMENTED** [review]
Overall, I think this is a good idea. 

Besides `LogSink`, do other sink clients like `stan`, `kafka` respect the way of putting metadata in `context` and get them passed down?

**Comment by whynowy** [comment]
And the other question I have is, if we have a `kafka` source, should we use the same `source` and `ID` for the whole pipeline, or we regenerate another set in next step?

File: runner/sidecar/out.go (hunk 1 above)
Comment (alexec) [reply runner/sidecar/out.go:43]: I'm not actually sure what we want to do with that. I know we want to be able to identify messages, for tracing purposes. I think we might want to transmit it downstream (i.e. add to meta-data).

**Pull request merged**